viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -579,77 +579,58 @@ void SLAMServiceImpl::GetLatestSampledPointCloudMapString(
         }
     }
 
+    // Get data from painted surface in RGBA format
     auto painted_surface = painted_slices->surface.get();
     int width = cairo_image_surface_get_width(painted_surface);
     int height = cairo_image_surface_get_height(painted_surface);
-    // Get all pixels from the painted surface in RGBA format
-    auto data = cairo_image_surface_get_data(painted_surface);
-
-    // Total number of bytes in image (4 bytes per pixel)
-    int size_data = width * height * 4;
-
-    // Each pixel contains 4 bytes of information in RGBA format
-    // data_vect[i + 0] is the R channel
-    // data_vect[i + 1] is the B channel
-    // data_vect[i + 2] is the G channel
-    // data_vect[i + 3] is the A channel
-    std::vector<unsigned char> data_vect(data, data + size_data);
-
+    auto image_data = cairo_image_surface_get_data(painted_surface);
+    // // Each pixel contains 4 bytes of information in RGBA format
+    // // data_vect[i + 0] is the R channel
+    // // data_vect[i + 1] is the B channel
+    // // data_vect[i + 2] is the G channel
+    // // data_vect[i + 3] is the A channel
+
+    // Iterate over image data and add to pointcloud buffer
     int num_points = 0;
-
-    // Sample the image based on the number of pixels. Output is the number of
-    // pixels to skip. skip_count will reduce the size of the PCD to under 32
-    // MB, with additional tuning provided by the samplingFactor. If the PCD
-    //  would already be smaller than 32MB, do not sample the image.
-    // When moving to streaming this behavior may change.
-    int skip_count = (size_data * pixelBytetoPCDByte) / maximumGRPCByteLimit *
-                     samplingFactor;
-    if (skip_count == 0) {
-        skip_count = 1;
-    }
-
     std::string data_buffer;
```

Comment:
Could we rename this to `pcd_data` to make it slightly more clear what it contains?

Reply:
Done!